port_ocean/clients/port/mixins/entities.py

```diff
@@ -124,38 +164,145 @@ async def upsert_entity(
+        ```
+        :param blueprint: The blueprint of the entities to be upserted
+        :param entities: A list of Entities to be upserted
+        :param request_options: A dictionary specifying how to upsert the entity
+        :param user_agent_type: a UserAgentType specifying who is preforming the action
+        :param should_raise: A boolean specifying whether the error should be raised or handled silently
+        :return: A list of tuples where each tuple contains:
+            - First value: True if entity was created successfully, False if there was an error, None if there was an error and the entity use search identifier
+            - Second value: The original entity (if failed) or the reduced entity with updated identifier (if successful)
+        :return: httpx.HTTPStatusError if there was an HTTP error and should_raise is False
+        """
+        validation_only = request_options["validation_only"]
+        async with self.semaphore:
+            logger.debug(
+                f"{'Validating' if validation_only else 'Upserting'} {len(entities)} of blueprint: {blueprint}"
+            )
+            headers = await self.auth.headers(user_agent_type)
+            response = await self.client.post(
+                f"{self.auth.api_url}/blueprints/{blueprint}/entities/bulk",
+                json={
+                    "entities": [
+                        entity.dict(exclude_unset=True, by_alias=True)
+                        for entity in entities
+                    ]
+                },
+                headers=headers,
+                params={
+                    "upsert": "true",
+                    "merge": str(request_options["merge"]).lower(),
+                    "create_missing_related_entities": str(
+                        request_options["create_missing_related_entities"]
+                    ).lower(),
+                    "validation_only": str(validation_only).lower(),
+                },
+                extensions={"retryable": True},
+            )
+        if response.is_error:
+            logger.error(
+                f"Error {'Validating' if validation_only else 'Upserting'} "
+                f"{len(entities)} entities of "
+                f"blueprint: {blueprint}"
+            )
+            handle_port_status_code(response, should_raise)
+            return httpx.HTTPStatusError(
+                f"HTTP {response.status_code}",
+                request=response.request,
+                response=response,
+            )
+        handle_port_status_code(response, should_raise)
+        result = response.json()
+        index_to_entity = {i: entity for i, entity in enumerate(entities)}
 
-        Returns:
-            Entity: A new entity with only the essential data
+        return self._parse_upsert_entities_batch_response(index_to_entity, result)
+
+    def _parse_upsert_entities_batch_response(
+        self,
+        index_to_entity: dict[int, Entity],
+        result: dict[str, Any],
```

Comment:
can we define what is expected structure within? e.g. (`entities: list[], errors: list[]`)